core/js/src/main/scala/cats/effect/unsafe/PolyfillExecutionContext.scala

```
@@ -47,11 +48,16 @@ private[unsafe] object PolyfillExecutionContext extends ExecutionContext {
           var postMessageIsAsynchronous = true
           val oldOnMessage = js.Dynamic.global.onmessage
 
-          js.Dynamic.global.onmessage = { () => postMessageIsAsynchronous = false }
-
-          js.Dynamic.global.postMessage("", "*")
-          js.Dynamic.global.onmessage = oldOnMessage
-          postMessageIsAsynchronous
+          try {
+            // This line throws `ReferenceError: onmessage is not defined` in JSDOMNodeJS environment
+            js.Dynamic.global.onmessage = { () => postMessageIsAsynchronous = false }
+            js.Dynamic.global.postMessage("", "*")
+            js.Dynamic.global.onmessage = oldOnMessage
+            postMessageIsAsynchronous
+          } catch {
+            case NonFatal(_) =>
```

This isn't going to work. Scala.JS (unfortunately) throws a fatal exception when referencing something in JS-land that isn't there at runtime. You've got to catch all exceptions.

---

Thanks again for your eyes on this ... currently this is passing in all test environments, is there something I'm missing??

---

No it turns out I'm wrong, it seems I'm the one that's missing something :smile: I tested this out and it works just as you're expecting it to.... which is so strange to me because my beliefs were built on experiences. Maybe Scala.JS has received some changes as to how this bit of surface area works and I must've missed it. I'll invalid my mental cache. Thanks for the collaboration :)

---

No worries! It sounded plausible to me too, I could have sworn I had similar experiences 🙃 But on the other hand I was able to replicate an issue with CI here, and make it go away, and it was consistent with your import scripts trick.